encode.go

```diff
@@ -632,6 +649,75 @@ type encMode struct {
 
 var defaultEncMode, _ = EncOptions{}.encMode()
 
+// These four decoding modes are used by getMarshalerDecMode.
+// maxNestedLevels, maxArrayElements, and maxMapPairs are
+// set to max allowed limits to avoid rejecting Marshaler
+// output that would have been the allowable output of a
+// non-Marshaler object that exceeds default limits.
+var (
+	marshalerForbidIndefLengthForbidTagsDecMode = decMode{
+		maxNestedLevels:  maxMaxNestedLevels,
+		maxArrayElements: maxMaxArrayElements,
+		maxMapPairs:      maxMaxMapPairs,
+		indefLength:      IndefLengthForbidden,
+		tagsMd:           TagsForbidden,
+	}
```

Comment:
Would it be better to avoid adding four new package-level variables for this?

A single array or a struct with four DecMode would remain a single package-level variable even if more DecMode are added later on.

Reply:
Good point!  Since these are not exported, this can be refactored without breaking changes when I have more bandwidth.